fdbserver/BlobManager.actor.cpp

```diff
@@ -5585,21 +5586,30 @@ ACTOR Future<Void> truncateMutationLogs(Reference<BlobManagerData> bmData) {
 	}
 }
 
+ACTOR Future<Reference<BlobConnectionProvider>> initManifestStore(Reference<BlobManagerData> bmData) {
+	std::string manifestUrl = wait(runRYWTransaction(bmData->db, [=](Reference<ReadYourWritesTransaction> tr) {
+		tr->setOption(FDBTransactionOptions::ACCESS_SYSTEM_KEYS);
+		tr->setOption(FDBTransactionOptions::LOCK_AWARE);
+		tr->setOption(FDBTransactionOptions::PRIORITY_SYSTEM_IMMEDIATE);
+		return BlobGranuleBackupConfig().manifestUrl().getD(tr);
+	}));
```

Comment:
```
	std::string manifestUrl = wait(runRYWTransaction(bmData->db, [=](Reference<ReadYourWritesTransaction> tr) {
		tr->setOption(FDBTransactionOptions::ACCESS_SYSTEM_KEYS);
		tr->setOption(FDBTransactionOptions::LOCK_AWARE);
		tr->setOption(FDBTransactionOptions::PRIORITY_SYSTEM_IMMEDIATE);
		return BlobGranuleBackupConfig().manifestUrl().getD(tr);
	}));
```

There is a helper wrapper of Database that can create transactions for RunTransaction and set these options, and a function to instantiate one which behaves exactly as what you have here since this is a common pattern, it is called `SystemDBWriteLockedNow`.

You can actually simplify this twice:  First to
```
	std::string manifestUrl = wait(runRYWTransaction(SystemDBWriteLockedNow(bmData->db.getReference()), [=](Reference<ReadYourWritesTransaction> tr) {
		return BlobGranuleBackupConfig().manifestUrl().getD(tr);
	}));
```

And then even further because `getD()` actually accepts _either_ a Transaction or a Database reference through some `if constexpr` magic that where if called with a Database it recursively calls itself wrapped in `runTransaction()`. 

So you can actually simplify this further to
```suggestion
	std::string manifestUrl = wait(BlobGranuleBackupConfig().manifestUrl().getD(SystemDBWriteLockedNow(bmData->db.getReference()));
	}));
```

Reply:
fixed. thank you!